packages/order-experience-api/src/api/createController.ts

```diff
@@ -88,14 +90,26 @@ export class CreateController extends AbstractController<typeof requestSchema> {
       ...calculateTotalsFromItems({ items }),
     })
 
-    let paymentFiltersData = undefined
-    if (!!paymentFilters && paymentFilters.length > 0) {
-      paymentFiltersData = await savePaymentFiltersAdmin(paymentFilters)
-    }
+    const paymentFiltersData =
+      !!paymentFilters && paymentFilters.length > 0
+        ? await this.savePaymentFilters(orderData, paymentFilters)
+        : undefined
+
     const dto = new Data({
       ...orderData,
       paymentFilters: paymentFiltersData ? paymentFiltersData : [],
     })
     return this.created<any>(res, dto.serialize())
   }
+
+  private async savePaymentFilters(
```

Comment:
Rename this to saveOrderPaymentFilters, because filters can be saved to merchant aswell

Reply:
Sure, makes more sense in that case.